internal/appsec/remoteconfig_test.go

```diff
@@ -371,3 +359,334 @@ func TestRemoteActivationScenarios(t *testing.T) {
+			{
+				ID:      "rule-1",
+				Enabled: true,
+			},
+			{
+				ID:      "rule-2",
+				Enabled: false,
+			},
+		},
+	}
+	overrides2 := rulesFragment{
+		Overrides: []rulesOverrideEntry{
+			{
+				ID:      "rule-3",
+				Enabled: true,
+			},
+			{
+				ID:      "rule-4",
+				Enabled: false,
+			},
+		},
+	}
+	invalidOverrides := rulesFragment{
+		Overrides: []rulesOverrideEntry{
+			{
+				Enabled: false,
+			},
+			{
+				ID:      "crs-930-100",
+				Enabled: false,
+			},
+		},
+	}
+	ackStatus := genApplyStatus(true, nil)
+
+	for _, tc := range []struct {
+		name        string
+		updates     map[string]remoteconfig.ProductUpdate
+		expected    map[string]rc.ApplyStatus
+		updateError bool
+	}{
+		{
+			name:     "single/ack",
+			updates:  craftRCUpdates(map[string]rulesFragment{"overrides": overrides}),
+			expected: map[string]rc.ApplyStatus{"overrides": ackStatus},
+		},
+		{
+			name:     "single/error",
+			updates:  craftRCUpdates(map[string]rulesFragment{"invalid": invalidOverrides}),
+			expected: map[string]rc.ApplyStatus{"invalid": genApplyStatus(true, errors.New("invalid configuration payload"))},
+		},
+		{
+			name:     "multiple/ack",
+			updates:  craftRCUpdates(map[string]rulesFragment{"overrides": overrides, "overrides2": overrides2}),
+			expected: map[string]rc.ApplyStatus{"overrides": ackStatus, "overrides2": ackStatus},
+		},
+		{
+			name:    "multiple/single-error",
+			updates: craftRCUpdates(map[string]rulesFragment{"overrides": overrides, "invalid": invalidOverrides}),
+			expected: map[string]rc.ApplyStatus{
+				"overrides": genApplyStatus(false, nil),
+				"invalid":   genApplyStatus(true, errors.New("invalid configuration payload")),
+			},
+		},
+		{
+			name:        "multiple/all-errors",
+			updates:     craftRCUpdates(map[string]rulesFragment{"overrides": overrides, "invalid": invalidRules}),
+			updateError: true,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			Start(WithRCConfig(remoteconfig.DefaultClientConfig()))
+			defer Stop()
+
+			if !Enabled() {
+				t.Skip("AppSec needs to be enabled for this test")
+			}
+
+			statuses := activeAppSec.onRCUpdate(tc.updates)
+			if tc.updateError {
+				for _, status := range statuses {
+					require.NotEmpty(t, status.Error)
+					require.Equal(t, rc.ApplyStateError, status.State)
+				}
+			} else {
+				require.Len(t, statuses, len(tc.expected))
+				require.True(t, reflect.DeepEqual(tc.expected, statuses))
+			}
+		})
+	}
+}
+
+// TestWafUpdate tests that the WAF behaves correctly after the WAF handle gets updated with a new set of security rules
+// through remote configuration
+func TestWafRCUpdate(t *testing.T) {
+	override := rulesFragment{
+		Overrides: []rulesOverrideEntry{
+			{
+				ID:      "crs-913-120",
+				Enabled: true,
+				OnMatch: []string{"block"},
+			},
+		},
+	}
+
+	Start()
+	defer Stop()
+
+	if !Enabled() {
+		t.Skip("AppSec needs to be enabled for this test")
+	}
+
+	t.Run("toggle-blocking", func(t *testing.T) {
```

Comment:
I don't see the value of starting the full appsec module here, because the test is then only leveraging the resulting `activeAppSec.cfg.rulesManager` and directly calling libddwaf with it. So you could be using a local rules manager instead of the full appsec module.

Reply:
Fixed in cc5abf5b67e13